controllers/apps/transformer_cluster_service.go

```diff
@@ -56,12 +63,18 @@ func (t *clusterServiceTransformer) Transform(ctx graph.TransformContext, dag *g
 		return err
 	}
 
+	npServices, err := listAllNodePortServices(transCtx)
+	if err != nil {
```

Comment:
the cost is huge to list all service for each reconcile, do it at only needing nodePort